### DISCUSSION TITLE
motor.load() values seem wrong sometimes

### DISCUSSION THREAD

**Discussion by FLL-Team-24277** [pybricks/support#1809] [General]
When running this code using a Spike medium motor, I am seeing something strange.

```python
from pybricks.pupdevices import Motor
from pybricks.parameters import Port
from pybricks.tools import wait

myMotor = Motor(Port.B)

myMotor.run(speed=1000)

while True:
    print(myMotor.load())
    wait(10)
```

While the motor is spinning, I can manually add load to it using my fingers, and I can hear the motor slowing down and I can see the load printing in the terminal. The strange thing is, the load() maxes out at 199, but the motor is still spinning and I can still put even more load on the motor. All the way until the motor stops, it just stays at 199. However, if I decrease the speed, to say 200, it seems to be fine. It maxes out at 199 still, but it seems to coincide with the actual stopping of the motor. 

Strangely, Lego says the max torque is 18 Ncm (180 mNm), so the 199 value seems in the right zone, but clearly I can add more load, and the motor is still turning, so it IS capable of more torque than they are documenting.

Is there anything I can do to get more of the full range of actual torque being used?

**Comment by BertLindeman** [comment]
Hello,

Trying your experiment, I see no real problem.

Tested on a primehub and a SPIKE Prime Medium Motor.
('primehub', '3.5.0', 'v1.20.0-23-g6c633a8dd on 2024-04-11')
```
loadA      0 stall False 
loadA     -4 stall False 
loadA   -199 stall False 
loadA   -196 stall False 
loadA   -199 stall True 
loadA   -199 stall False 
loadA   -145 stall False 
loadA     -2 stall False 
```
Braking the motor by hand it almost stands still on load -199.

Tested with other motortypes and all show negative load.
Different type of motors produce different loads. Obviously. 

Nothing scientific from me, hand brake the motors.

| motor type | load |
|-|-|
| Technic Large Angular motor, gray | -560 |
| Technic Control+ XL Motor | -233 |
| Technic Control+ Large Motor | -260 | 
| Technic Large Angular motor, gray | -560 |

I wonder if load should be reported negative.
What firmware did **you** test with?

Bert

**Reply by FLL-Team-24277** [reply]
Here's a video that shows what I am seeing. I think the reported load be able to go past the 199 mNm. I don't have a way to test what the ACTUAL load is, so maybe the load() function is just out of calibration. I don't know.

Version:
('primehub', '3.5.0', 'v1.20.0-23-g6c633a8dd on 2024-04-11')

https://youtu.be/amEX4KFCQMw

**Comment by FLL-Team-24277** [comment]
@dlech , do you have any thoughts on this?

**Reply by dlech** [reply]
DC motor torque depends on electrical power supplied and the speed of the motor. (Here is a random page I found that seems like it has a decent introduction: http://lancet.mit.edu/motors/motors3.html)

When requesting a specific speed, the motor control algorithm in Pybricks will regulate the voltage supplied to the motor to try to maintain the requested speed using a feedback loop. When there is no load on the motor, the voltage has to be reduced to slow down the motor to maintain that speed. But as you apply more load to the motor, more and more voltage has to be supplied in order to create enough torque for that load. Eventually, we reach the maximum available voltage (the battery voltage). After that, any additional load will cause the motor to slow down to less that the requested speed since we can't supply more power. 

So using speed control to measure the torque characteristics of the motor isn't the best way to go about it. Instead, I would suggest using the `Motor.dc()` function to deliver a constant voltage to the motor. And also print the speed so that you can see that part of the equation.

Finally, there isn't a way to measure the electrical current to each motor on any of the LEGO hubs, so we can't actually accurately measure the power being delivered to the motor. So the torque returned by the `Motor.load()` function is just an estimate based on a model of the motor characteristics and isn't going to be super-accurate.

**Reply by laurensvalk** [reply]
To add to this:

When running at a controlled speed as in your example, the hub estimates the motor load as a measure of how hard it has to work to minimize deviations from that speed. This is why it maxes out at some point - that's the most the motor can deliver. Given the tools we have, this is a reasonably good metric for quite a lot of applications, despite not being entirely exact.

**Comment by FLL-Team-24277** [comment]
Thanks! That is very helpful. At least I know it's not something I am doing wrong. I do think I will take a look at dc(), but I think load() will work good enough for us.
